pymatgen/io/lammps/inputs.py

```diff
@@ -46,58 +150,54 @@ def get_string(self):
 
         """
 
-        lammps_input = '## LAMMPS input generated from LammpsInputConstructor\n\n'
+        lammps_input = "## LAMMPS input generated from LammpsInputConstructor\n\n"
         for stage, command_dict in self.input_settings.items():
             for command, args in command_dict.items():
-                if command.startswith('comment') or command.startswith('header'):
-                    if command.startswith('header'):
-                        lammps_input += '\n'
-                    lammps_input += self.add_string(args)
+                if command.startswith("comment") or command.startswith("header"):
+                    if command.startswith("header"):
+                        lammps_input += "\n"
+                    lammps_input += args + "\n"
                 else:
-                    lammps_input += self.format_command(command, args)
+                    lammps_input += f"{command:20} {args}\n"
         return lammps_input
 
-    @staticmethod
-    def format_command(command, args):
+    def from_string(self, s: str):  # type: ignore
         """
-        Simple format helper for adding command with the arguments.
+        Helper method to parse string representation of LammpsInputFile
 
         Args:
-            command: LAMMPS command.
-            args: Arguments for LAMMPS command.
-
-        Returns:
+            s: String representation of LammpsInputFile.
 
+        Returns: LammpsInputFile
         """
 
-        return "{0:20} {1}\n".format(command, args)
+        for line in self._clean_lines(s.splitlines()[1:]):
+            if line[:2] == "##":
+                self.add_comment(comment=line[2:], is_stage_header=False)
+            if line[0] == "#":
+                self.init_stage()
+                self.add_comment(comment=line[2:], is_stage_header=True)
+            else:
+                self.add_commands(line)
 
-    @staticmethod
-    def add_string(string):
-        """
-        Helper method for adding string to lammps input file.
-        Args:
-            string: String to be added to lammps input file.
-        """
-        # return '\n' + string + '\n'
-        return string + '\n'
+        return LammpsInputFile(self.input_settings)
```

Comment:
OK, this method cannot reference `self` because it's purpose is to _construct_ a `LammpsInputFile` from a string. So there is no `self` until this method returns it. Right now your method is written as if a `LammpsInputFile` already exists and you want to update it.

I see two approaches you could take
1. Create an empty `LammpsInputFile`, then `add_comment` / `add_commands` to update it, and return that object
2. Use string parsing to build an `OrderedDict`, then init your `LammpsInputFile` at the end and return that

If you use Option 2, you would need to make `_clean_lines`, `add_comment`, and `add_commands` static methods. That makes sense for `clean_lines` b/c it's just a utility function that has nothing to do with the contents of this class, but `add_comment` and `add_commands` make less sense as static methods.